core/src/avm1/globals/bitmap_data.rs

```diff
@@ -800,13 +800,90 @@ pub fn copy_pixels<'gc>(
 }
 
 pub fn merge<'gc>(
-    _activation: &mut Activation<'_, 'gc, '_>,
+    activation: &mut Activation<'_, 'gc, '_>,
     this: Object<'gc>,
-    _args: &[Value<'gc>],
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
     if let Some(bitmap_data) = this.as_bitmap_data_object() {
         if !bitmap_data.disposed() {
-            log::warn!("BitmapData.merge - not yet implemented");
+            let source_bitmap = args
+                .get(0)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_object(activation);
+
+            let source_rect = args
+                .get(1)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_object(activation);
+
+            let src_min_x = source_rect
+                .get("x", activation)?
+                .coerce_to_i32(activation)?;
+            let src_min_y = source_rect
+                .get("y", activation)?
+                .coerce_to_i32(activation)?;
+            let src_width = source_rect
+                .get("width", activation)?
+                .coerce_to_i32(activation)?;
+            let src_height = source_rect
+                .get("height", activation)?
+                .coerce_to_i32(activation)?;
+
+            let dest_point = args
+                .get(2)
+                .unwrap_or(&Value::Undefined)
+                .coerce_to_object(activation);
+
+            let dest_x = dest_point.get("x", activation)?.coerce_to_i32(activation)?;
+            let dest_y = dest_point.get("y", activation)?.coerce_to_i32(activation)?;
```

Comment:
`coerce_to_i32` is specifically for when there is wrapping behavior, which is incorrect in this case:
```
b2.merge(b, new Rectangle(4294967296,0,200,200), new Point(4294967296, 0), 256.0, 0.0, 0.0, 0.0);
```
Probably want `coerce_to_f32(..)? as i32`. I guess this applies to all of the other fns with `sourceRect`/`destPoint`, sorry I didn't notice earlier.

Reply:
There's no `coerce_to_f32`, so I used `coerce_to_f64` instead.

> sorry I didn't notice earlier.

No problem, better late than never! :)
I should have been the one to check which one is correct.